src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -79,6 +79,25 @@
    default-jvm-metrics-allowed
    http-client-metrics-allowed-hists))
 
+(defn jdk-support-status
+  "Returns :official, :deprecated, :unknown, or :unsupported."
+  [version]
+  (cond
+    (re-matches #"1\.[1234567]($|(\..*))" version) :unsupported
```

Comment:
I had no idea that you could alternate a line ending like this and would have done something like
```
"1\.[1-7](\..*)?$"
```
That's pretty cool!